modules/ROOT/pages/instanton-limitations.adoc

```diff
@@ -161,6 +162,72 @@ This configuration uses placeholder values for things like the database name, ho
 
 If an application is injected with a `DataSource` before the checkpoint and the configuration of the `DataSource` changes, the application is restarted when the InstantOn application container image is run with the updated configuration. You can avoid this scenario by using the `beforeAppStart` option or by modifying the component not to be early startup code. In this example, that modification is to remove the `loadOnStartup = 1` attribute.
 
+[#mp-config-no-default]
+== Accessing microprofile config with no default value at Checkpoint
+The configuration property can be introduced inside the application either statically through `<T>` or `Optional<T>`, or dynamically using a `Provider<T>`. The following example shows ways to inject static, static-optional, dynamic and dynamic-optional config property. 
+[source,java]
+----
+  @Inject
+  @ConfigProperty(name = "static_config")
+  String staticConfig;
+
+  @Inject
+  @ConfigProperty(name = "static_optional_config")
+  Optional<String> staticOptionalConfig;
+
+  @Inject
+  @ConfigProperty(name = "dynamic_config")
+  Provider<String> dynamicConfig;
+
+  @Inject
+  @ConfigProperty(name = "dynamic_optional_config")
+  Provider<Optional<String>> dynamicOptionalConfig;
+----
+The injected `static_config` will cause an error like the following example if there is no value found in an existing config source during checkpoint.
+
+[source,sh]
+----
+SRCFG02000: Failed to Inject @ConfigProperty for key static_config into io.example.Example.staticConfig since the config property could not be found in any config source.
+----
+
+This error can be avoided by providing a default value for the configuration key in one of the following ways:
+
+Specify the default value on the @ConfigProperty annotation like the following:
+[source,java]
+----
+  @Inject
+  @ConfigProperty(name = "static_config", defaultValue = "defaultValue")
+  String staticConfig;
+----
+
+Specify the default value in the application `META-INF/microprofile-config.properties` resource.
+[source,sh]
+----
+  static_config=defaultValue
+----
+
+Specify a default value in a server.xml variable.
+[source,xml]
+----
+  <variable name="static_config" defaultValue="defaultValue" />
+----
+
+The `defaultValue` can then be overridden on restore by configuring the value with an environment variable or values from the variable source directory.
+
+However, if no default value is set, we can still avoid the above error by injecting config using `static_optional_config`, `dynamic_config`, `dynamic_optional_config`. But caution is needed for CDI beans that are `@ApplicationScoped` which are created during application startup when using the afterAppStart checkpoint option. Trying to access the `dynamic_config` early during the application startup at checkpoint can cause the following error:
```

Comment:
so does the error that was initially described at line 190 only applies to static required config? As, written, It's hard to tell whether the error resulting from lack of default value is dependent on how you inject it, or simply avoidable by having multiple options to inject.